clean_code_main/clean_code_checks/y_check_aunit_coverage_check.clas.abap

```diff
@@ -0,0 +1,375 @@
+      RETURN.
+    ENDIF.
+
+    DATA(unit_test_result) = get_unit_test_result( main_object = main_object ).
+
+    process_unit_test_result( unit_test_result = unit_test_result ). "Return the unit test coverage to the SCI framework
+
+  ENDMETHOD.
+
+
+  METHOD constructor.
+
+    """"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+    " Set up the check: Put check in the right folder, provide a name and a version
+    """"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+    super->constructor( ).
+    description    = 'Unit test coverage'(000).  "The name, under which this check will appear in SCI
+    category       = 'Y_CHECK_CATEGORY'.         "The folder, where this check will be located in
+    version        = '001'.
+
+    settings-pseudo_comment = '"#EC UNITTEST_MISSING' ##NO_TEXT.
+    attributes_ok = abap_true.      "Default values are okay for instant usage
+    has_documentation = abap_true.  "This check features a documentation that can be viewed in SCI
+    settings-documentation = |{ c_docs_path-checks }abap-unit-coverage.md|.
+    has_attributes = abap_true.     "This check features parameters, that can be set by the user in SCI
+
+    DATA(all_messages_of_this_check) = VALUE scimessages(
+      ( test = myname code = 'gen' kind = c_error text = 'Source code does not have unit test or test can not be generated'(gen) )
+      ( test = myname code = 'fal' kind = c_error text = 'Execution of the unit tests failed'(fal) )
+      ( test = myname code = 'tlw' kind = c_error text = 'Unit test coverage too low: &1%'(tlw) )
+      ( test = myname code = 'low' kind = c_warning text = 'Unit test coverage low: &1%'(low) )
+      ( test = myname code = 'iok' kind = c_note text = 'Unit test coverage ok: &1%'(iok) )
+      ( test = myname code = 'eas' kind = c_note text = 'Source code object has a low complexity. Unit tests are not mandatory.'(eas) ) ).
+    INSERT LINES OF all_messages_of_this_check INTO TABLE scimessages.
+
+    "The SAP standard check class for cyclomatic complexity will raise an event MESSAGE,
+    "if the cyclomatic complexity is exceeded for a given object
+    complexity_checker = NEW cl_ci_test_metric_proc( ).
+    SET HANDLER check_unit_test_coverage FOR complexity_checker.
+
+  ENDMETHOD.
+
+
+  METHOD GET_ATTRIBUTES.
+
+    """""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+    " Write user-defined parameters for this check to a memory object - Will be needed later
+    """""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+    EXPORT
+      complexity = threshold_for_complexity
+      yellow_alert = threshold_for_yellow_alert
+      red_alert = threshold_for_red_alert
+      TO DATA BUFFER p_attributes.
+
+  ENDMETHOD.
+
+
+  METHOD GET_CYCLOMATIC_COMPLEXITY.
```

Comment:
why cyclomatic complexity?

Reply:
Developers should not be forced to produce unit tests for boiler plate code, GETter/SETers and empty methods. This would be a waste of time and would not be helpful. Thats why the check is only carried out for objects above a specific complexity.